src/components/Link/Link.scss

```diff
@@ -49,3 +49,10 @@
 .ms-Link {
   @include ms-Link;
 }
+
+// For button links we need to ensure by default that we inherit the font size. Otherwise
+// the button default font size will override.
+
+BUTTON.ms-Link {
+  font-size: inherit;
```

Comment:
Just a correction: it's not that the browser styles are more specific, it's that they take precedence as styles applied directly to the element over inherited styles. (Specificity doesn't really apply to inherited styles.)